src/main/java/seedu/duke/logic/preparecommand/PrepareAddSetCommand.java

```diff
@@ -33,16 +43,26 @@ public Command prepareCommand() {
         Date date = new Date();
         SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
         String strDate = formatter.format(date);
+        String initialPath = new File("").getAbsolutePath();
+        String filePath = initialPath + "/" + description[1] + ".txt";
+
         try {
-            String initialPath = new File("").getAbsolutePath();
-            String filePath = initialPath + "/" + description[1] + ".txt";
             File file = new File(filePath);
             BufferedReader reader = new BufferedReader(new FileReader(filePath));
 
             if (file.exists()) {
                 String line = reader.readLine();
+                checkEmptyFile(filePath);
 
                 while (line != null) {
+                    checkTags(line);
+                    String description = line.substring(2, line.indexOf("c/") - 1);
+                    String calories = line.substring(line.indexOf("c/") + 2);
```

Comment:
can consider making these magic numbers final and if i recall correctly i have defined this as CALORIE_TAG